components/brave_rewards/core/rewards_engine_impl.cc

```diff
@@ -577,27 +578,33 @@ void RewardsEngineImpl::FetchBalance(FetchBalanceCallback callback) {
   });
 }
 
-void RewardsEngineImpl::GetExternalWallet(const std::string& wallet_type,
-                                          GetExternalWalletCallback callback) {
-  WhenReady([this, wallet_type, callback = std::move(callback)]() mutable {
-    if (wallet_type == constant::kWalletBitflyer) {
-      return bitflyer()->GetWallet(std::move(callback));
+void RewardsEngineImpl::GetExternalWallet(GetExternalWalletCallback callback) {
+  WhenReady([this, callback = std::move(callback)]() mutable {
+    auto wallet_type = GetState<std::string>(state::kExternalWalletType);
```

Comment:
For profiles created before July 2021 (this is when Gemini was added), the `external_wallet_type` pref doesn't exist for bitFlyer and Uphold users. For them the `GetExternalWallet()` flow would return `nullptr`.

Reply:
Thanks for the reminder! I'll add the necessary logic (possibly a migration) and document.

Reply:
Migration added